simulationworkflowschema/equation_of_state.py

```diff
@@ -158,9 +222,75 @@ def normalize(self, archive, logger):
             self.results = EquationOfStateResults()
             self.outputs.append(Link(name=WORKFLOW_RESULTS_NAME, section=self.results))
 
-        if not self._calculations:
+        try:
+            task_archives = [task.task.m_root() for task in self.tasks]
+            assert all(
+                isinstance(task_archive.workflow2, SinglePoint)
+                for task_archive in task_archives
+            )
+        except Exception:
+            logger.warning(
+                'Not all tasks are SinglePoints or failed to retrieve task archives. EOS workflow may be incomplete or incorrect.'
+            )
             return
 
+        for task in self.tasks:
+            # DEBUGGING: resolve task.inputs their its actual object
+            # in order to test non-proxy case with proxy data
+            # TODO - Remove this after proper testing is included
+            # task_input_ids = [
+            #     id(
+            #         inp.section.m_resolved()
+            #         if isinstance(inp.section, MProxy)
+            #         else inp.section
+            #     )
+            #     for inp in task.inputs
+            # ]
+            # Check for matches by proxy value
+            input_proxy_values = [input.section.m_proxy_value for input in task.inputs]
+            if input_proxy_value in input_proxy_values:
+                # get the index of the match
+                index = input_proxy_values.index(input_proxy_value)
+                task.inputs[index].name = input_name
+            else:
+                # TODO - Test this!
+                # add the input structure to each task if not already present
+                task.inputs.append(Link(name=input_name, section=input_section))
+
+            # ! Does not work since I want to match simply the archive path of the section
+            # Check for matches by section id
```

Comment:
I still do not know how to compare the archive path of the global input and the task input. Some other approach would be needed for the case that the task.inputs[].section are not proxies

Reply:
ignore this, m_proxy_value works in both cases, I was just confused by this weird old test data